plugins/NebulaTextures/src/gui/NebulaTexturesDialog.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Nebula Textures plug-in for Stellarium
+ *
+ * Copyright (C) 2024-2025 WANG Siliang
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef NEBULATEXTURESDIALOG_HPP
+#define NEBULATEXTURESDIALOG_HPP
+
+#include "StelDialog.hpp"
+#include <QString>
+#include <QPair>
+
+#include "NebulaTextures.hpp"
+#include "StelSkyLayerMgr.hpp"
+#include "StelApp.hpp"
+
+#define MS_CONFIG_PREFIX QString("NebulaTextures")
+#define CUSTOM_TEXNAME QString("Custom Textures")
+#define DEFAULT_TEXNAME QString("Nebulae")
+#define TEST_TEXNAME QString("Test Textures")
+
+class Ui_nebulaTexturesDialog;
+
+//! Main window of the Nebula Textures plug-in.
+//! @ingroup nebulaTextures
+class NebulaTexturesDialog : public StelDialog
+{
+	Q_OBJECT
+
+public:
+	NebulaTexturesDialog();
+	~NebulaTexturesDialog() override;
+
+   StelSkyImageTile* get_aTile(QString key);
+
+public slots:
+	void retranslate() override;
+
+
```

Comment:
please add source documentation at least to the non-trivial functions. What does it do, what are the arguments, ...